gustaf/spline/create.py

```diff
@@ -0,0 +1,229 @@
+"""gustaf/spline/create.py
+
+Create operations for spline geometries
+"""
+
+
+import itertools
+import logging
+import numpy as np
+
+
+def extrude(spline, axis=None):
+    """
+    Extrudes Splines ype objects of any spline-type object
+
+    Parameters
+    ----------
+    spline : Gustaf-Spline
+    axis : Extrusion axis
+    """
+    from gustaf.spline.base import GustafSpline
+
+    # Check input type
+    from gustaf.spline.base import GustafSpline
+    if not issubclass(type(spline), GustafSpline):
+        raise NotImplementedError("Extrude only works for splines")
+
+    # Check axis
+    if axis is not None:
+        axis = np.asarray(axis)
+        if len(axis.shape) == 0:
+            axis = np.asarray([axis])
+    else:
+        raise ValueError("No extrusion Axis given")
+
+    # Check Axis dimension
+    if not (spline.control_points.shape[1] == axis.shape[0]):
+        raise ValueError(
+            "Dimension Mismatch between extrusion axis and spline."
+        )
+
+    # Start Extrusion
+    arguments = {}
+
+    arguments["degrees"] = np.concatenate((spline.degrees, [1]))
+    arguments["control_points"] = np.concatenate((
+        spline.control_points,
+        spline.control_points + axis))
+    if "knot_vectors" in spline._required_properties:
+        arguments["knot_vectors"] = spline.knot_vectors + [[0,0,1,1]]
+    if "weights" in spline._required_properties:
+        arguments["weights"] = np.concatenate((spline.weights, spline.weights))
+
+    return type(spline)(**arguments)
+
+def revolve(spline, axis=None, center=None, angle=None, n_knot_spans=None):
+    """
+    Revolve spline around an axis and extend its parametric dimension
+
+    Parameters
+    ----------
+    spline : GustafSpline
+      Basis-Spline to be revolved
+    axis : np.ndarray
+      Axis of revolution
+    center : np.ndarray
+      Center of revolution
+    angle : float
+    n_knot_spans : int
+      number of non-zero knot-elements for result-spline (if applicable)
+
+    Returns
+    -------
+    spline : spline-type          
+    """
+    from gustaf.spline.base import GustafSpline
+
+    # Check input type
+    if not issubclass(type(spline), GustafSpline):
+        raise NotImplementedError("Revolutions only works for splines")
+
+    # The parametric dimension is independent of the revolution but the
+    # rotation-matrix is only implemented for 2D and 3D problems
+    if not (spline.dim == 2 or spline.dim == 3):
+        raise NotImplementedError(
+            "Revolutions only implemented for 2D and 3D geometries"
```

Comment:
```suggestion
            "Sorry,"
            "revolutions only implemented for 2D and 3D splines"
```
We gently apologize, since this is not N-dim